### PR TITLE
Refactor the code so the LibraryViewController.cs can be tested

### PR DESCRIPTION
**Pull request by monikaprabhu**
Move methods to separate class "Event Controller" , so this class can be used for mock testing.
Derived class object LibraryViewController needs parameters which cannot be mocked. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### @sharadkjaiswal 
Please review and merge the PR.

### CODE REVIEW

**Comment by monikaprabhu** [comment]
FYI.... @Benglin

File: src/LibraryViewExtension/LibraryViewController.cs [hunk 1]
```diff
@@ -60,6 +104,8 @@ public LibraryViewController(Window dynamoView, ICommandExecutive commandExecuti
             InitializeResourceStreams(dynamoViewModel.Model);
         }
 
+        public LibraryViewController() { }
```
Comment (sharadkjaiswal) [inline src/LibraryViewExtension/LibraryViewController.cs]: This constructor is not needed, you can remove it.

File: test/ViewExtensionLibraryTests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (sharadkjaiswal) [inline test/ViewExtensionLibraryTests/Properties/AssemblyInfo.cs]: you can reference sharedAssemblyInfo.cs to get the standard versioning for Dynamo dll as it is done for other projects.

File: src/Dynamo.All.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.25420.1
+# Visual Studio 2013
```
Comment (sharadkjaiswal) [inline src/Dynamo.All.sln]: you can revert changes in this portion to keep it compatible with VS2015

**Comment by sharadkjaiswal** [comment]
Thanks Monika, apart from few minor comments this looks good.

File: test/ViewExtensionLibraryTests/ViewExtensionLibraryTests.csproj [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    <Optimize>false</Optimize>
+    <OutputPath>$(OutputPath)</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (sharadkjaiswal) [inline test/ViewExtensionLibraryTests/ViewExtensionLibraryTests.csproj]: you need to update this as well to output to $(OutputPath)

**Comment by monikaprabhu** [comment]
@sharadkjaiswal - I've resolved the PR comments above please review and merge.

**Pull request merged**